### DISCUSSION TITLE
C++ example for flatbuffer and mcap

### DISCUSSION THREAD

**Discussion by andrewwongagility** [foxglove/mcap#1090] [General] [answered]
I'm wondering if there are any C++ code examples for writing an mcap file with flatbuffers?

I'm looking for something similar to the nicely written example for [using mcap with protobuf](https://mcap.dev/guides/cpp/protobuf#writing).

Thanks, any help is greatly appreciated!

**Comment by jtbandes** [comment, marked as answer]
We don’t have a C++ MCAP FlatBuffers example right now but you can look at these for inspiration:
- typescript/mcap/flatbuffers https://github.com/foxglove/mcap/blob/db7675fedc1c2da9d1b9e3a58efc7074a43dfdcd/typescript/examples/flatbufferswriter/scripts/main.ts
- C++/websocket/flatbuffers https://github.com/foxglove/ws-protocol/blob/main/cpp/examples/src/example_server_flatbuffers.cpp

**Comment by andrewwongagility** [comment]
Got it, thank you!
